pkg/apiserver/resthandler.go

```diff
@@ -299,6 +301,27 @@ func createHandler(r rest.NamedCreater, scope RequestScope, typer runtime.Object
 			return
 		}
 
+		unstruct := &runtime.Unstructured{}
+		err = runtime.UnstructuredJSONScheme.DecodeInto(body, unstruct)
```

Comment:
It's not "bad" -- it's more or less what I suggested. However, I'm concerned about the performance impact and whether the re-marshaling could cause any issues. Refactoring Decode doesn't look too difficult -- it starts by extracting the apiVersion and kind in a similar manner.